src/redux/utils/connect-extension-middleware.ts

```diff
@@ -10,71 +9,104 @@ import { isEqual } from 'lodash';
         }
-        if (!isEqual(preferences, lastSentState?.preferences)) {
-            statePatch.preferences = preferences;
-        }
-        if (!isEqual(contacts, lastSentState?.contacts)) {
-            statePatch.contacts = contacts;
+
+        const trimmedContacts = trimContacts(contacts);
+        if (!isEqual(trimmedContacts, lastSentState?.trimmedContacts)) {
+            statePatch.trimmedContacts = trimmedContacts;
         }
-        if (!isEqual(tokens, lastSentState?.tokens)) {
-            statePatch.tokens = tokens;
+
+        const trimmedPreferences = trimPreferences(preferences);
+        if (!isEqual(trimmedPreferences, lastSentState?.trimmedPreferences)) {
+            statePatch.trimmedPreferences = trimmedPreferences;
         }
 
-        if (!isEqual(tokens, lastSentState?.tokens)) {
-            statePatch.tokens = tokens;
+        const trimmedTokens = trimTokens(tokens);
+        if (!isEqual(trimmedTokens, lastSentState?.trimmedTokens)) {
+            statePatch.trimmedTokens = trimmedTokens;
         }
 
         if (statePatch) {
-            lastSentState = { ...lastSentState, ...statePatch };
+            lastSentState = {
+                ...lastSentState,
+                ...statePatch
+            };
         }
 
         return statePatch;
     }
 );
 
+let timer;
+const CONN_EXTENSION_STATE = `${CONN_EXTENSION}State`;
+
 export const connectExtensionMiddleware = store => next => async action => {
     next(action);
 
-    if (Platform.OS !== 'web' && (await ConnectExtensionWeb.isConnected())) {
-        const statePatch = getStatePatch(store.getState());
+    timer && clearTimeout(timer);
+
+    timer = setTimeout(async () => {
+        if (Platform.OS !== 'web' && (await ConnectExtensionWeb.isConnected())) {
+            const statePatch = getStatePatch(store.getState());
+
+            const currentStateHash = await hash(JSON.stringify(statePatch));
 
-        if (Object.keys(statePatch).length > 0) {
             try {
-                const connection = await ConnectExtensionWeb.getConnection();
+                const storageStateHash = await getItemFromStorage(CONN_EXTENSION_STATE);
 
-                if (connection) {
-                    await ConnectExtension.syncExtension(connection);
+                if (storageStateHash) {
+                    if (storageStateHash !== currentStateHash) {
+                        // Sync Extension with the new state
+                        if (Object.keys(statePatch).length > 0) {
+                            try {
+                                const connection = await ConnectExtensionWeb.getConnection();
+
+                                if (connection) {
+                                    await ConnectExtension.syncExtension(connection);
+                                }
+                            } catch {
+                                lastSentState = {};
+                            }
+                        }
+                    }
```

Comment:
If there is a storageStateHash and is not equal to the currentState you need to update it after you sync it

Reply:
Yes true, fixed.